### PR TITLE
Fixes #30858 - bump rubocop to 0.88

### PR DESCRIPTION
**Pull request by lzap**
All the gems supports Ruby 2.4 or higher according to info from
rubygems.org.

### CODE REVIEW

**Comment by lzap** [comment]
I created `.rubocop_new.yml` file which holds new cops reported by the tool as pending: https://docs.rubocop.org/rubocop/versioning.html#pending-cops

We can continue with slow approach of enabling them. I will make followup PRs with enabling some of them which are cheap to enable (not many changes, bigger benefits).

File: .rubocop_new.yml [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+Performance/SortReverse: # (new in 1.7)
+  Enabled: false
+Performance/Squeeze: # (new in 1.7)
+  Enabled: false
+Performance/StringInclude: # (new in 1.7)
+  Enabled: false
+Performance/CollectionLiteralInLoop:
+  Enabled: false
+Performance/Sum: # (new in 1.8)
+  Enabled: false
```
Comment (lzap) [inline .rubocop_new.yml]: Most of Performance cops are small changes, I'd like to file PR for discussion.

File: .rubocop_new.yml [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+Rails/MatchRoute: # (new in 2.7)
+  Enabled: false
+Rails/NegateInclude: # (new in 2.7)
+  Enabled: false
+Rails/Pluck: # (new in 2.7)
+  Enabled: false
+Rails/PluckInWhere: # (new in 2.7)
+  Enabled: false
+Rails/UniqueValidationWithoutIndex:
+  Enabled: false
```
Comment (lzap) [inline .rubocop_new.yml]: This one (`UniqueValidationWithoutIndex`) reveals we have number of validations without unique indices.

**Comment by lzap** [comment]
> Some files could not be reviewed due to errors:

Damn is the reason why Rubocop is pinned because of Hound? It does not appear to understand those new cops.

**Comment by ekohl** [comment]
> Damn is the reason why Rubocop is pinned because of Hound? It does not appear to understand those new cops.

Yes, we are dependent on Hound.

**Comment by lzap** [comment]
I've decreased version to 0.88 which is the last version supported by Hound. Then I had to decrease every single plugin version one by one until Hound stopped complaining, their documentation does not state which version of plugins they support and it was complaining about various Cops as unknown. Now it should pass both on Hound and our CI. @ezr-ondrej @ekohl

**Comment by ezr-ondrej** [comment]
Thanks @lzap for taking care of this. I would say, that this is exactly the time to get rid of Hound though.

**Comment by lzap** [comment]
Well we can merge this and then get rid of Hound (in another PR) perhaps? I've invested quite some time into this.

Tests are passing, integration tests (JS) are failing.

**Comment by ezr-ondrej** [comment]
[test katello]
Agreed :+1:

**Review by ezr-ondrej — APPROVED** [review]
LGTM :+1:

**Comment by ezr-ondrej** [comment]
Stops Hound from complaining :+1: Once we know what to do about our RuboCop situation we can take it further.
Thanks @lzap !

**Comment by ezr-ondrej** [comment]
oh, right, it needs @theforeman/packaging approval.

**Comment by ezr-ondrej** [comment]
#7980 has been merged, closing this.

**Pull request closed without merging**